src/App.tsx

```diff
@@ -1,51 +1,103 @@
 import React from 'react';
 import './App.scss';
+import classNames from 'classnames';
 
-// import goodsFromServer from './goods';
-
-export const App: React.FC = () => {
-  return (
-    <main className="App">
-      <header className="App__header">
-        <h1 className="App__title">
-          Jam is selected
-        </h1>
-
-        <button
-          type="button"
-          className="App__clear"
-        >
-          Clear
-        </button>
-      </header>
-
-      <ul>
-        <li className="Good">Dumplings</li>
-        <li className="Good">Carrot</li>
-        <li className="Good">Eggs</li>
-        <li className="Good">Ice cream</li>
-        <li className="Good">Apple</li>
-        <li className="Good">Bread</li>
-        <li className="Good">Fish</li>
-        <li className="Good">Honey</li>
-        <li className="Good Good--active">Jam</li>
-        <li className="Good">Garlic</li>
-      </ul>
-
-      {/* Put required buttons into each Good */}
-      <button
-        type="button"
-        className="Good__remove"
-      >
-        Remove
-      </button>
-
-      <button
-        type="button"
-        className="Good__select"
-      >
-        Select
-      </button>
-    </main>
-  );
+import goodsFromServer from './goods';
+
+type State = {
+  selectedGoods: string[];
 };
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGoods: ['Jam'],
+  };
+
+  removeEl = (list: string[], good: string) => {
+    return list.filter((el: string) => el !== good);
+  };
+
+  titleDescription = (list: string[]) => {
+    switch (list.length) {
+      case 0:
+        return 'No goods selected';
+
+      case 1:
+        return `${list[0]} is selected`;
+
+      default:
+        return `${list.slice(0, -1).join(', ')} and ${list[list.length - 1]} are selected`;
+    }
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <main className="App">
+        <header className="App__header">
+          <h1 className="App__title">
+            {this.titleDescription(selectedGoods)}
+          </h1>
+
+          {selectedGoods.length
+            ? (
+              <button
+                type="button"
+                className="App__clear"
+                onClick={() => {
+                  this.setState({ selectedGoods: [] });
+                }}
+              >
```

Comment:
Let's create separate method for this, so we don't create new arrow function after component re-rendering